src/main/scala/lc/server/Server.scala

```diff
@@ -55,6 +60,14 @@ class Server(port: Int) {
     Response(405, write(message).getBytes)
   }
 
+  private def getResponse(response: ChallengeResult): Response = {
+    response match {
+      case Image(image) => Response(200, image)
+      case Error(_)     => Response(500, write(response).getBytes)
+      case _            => Response(200, write(response).getBytes)
+    }
+  }
+
```

Comment:
Just a hint: If you create separate class hierarchies for separate use-cases, you might need two functions to replace the above function.

Reply:
Didn't quite get this. Let's discuss this over a call.